crates/sui-graphql-rpc/src/mutation.rs

```diff
@@ -119,6 +120,10 @@ impl Mutation {
             })
             .collect();
 
+        let latest_cp = Checkpoint::query_latest_checkpoint_sequence_number(ctx.data_unchecked())
+            .await
+            .extend()?;
+
```

Comment:
> I don't think setting the consistent context for a just executed transaction to the latest checkpoint that we indexed is a coherent thing to do

While I agree with this, my rationale was that for the cursors on the dry run or write path, whether we set a null result or a checkpoint value doesn't matter since it is unlikely to be used again - a caller is not going to be paginating over something like the object changes on a transaction on a dry run or execution, and they should also be disincentivized from using that cursor on a read query.

On the read side, it’s more ergonomic to have this context value be non-null - since it is - rather than be an Option<T> to accommodate DryRun and ExecutionResult

Reply:
Like I said, I'm happy for it to not be optional, but I think setting it to the latest available checkpoint is both extra work and also incorrect in an observable way -- even if you can't paginate transaction effects today, you could still query for the transaction sender and further query its objects, or sent transaction blocks, which is subject to this context.

If we were able to serve that query correctly the transaction that was just executed would be in the list of transactions under the `sender` query, but it won't be if we set the context to the latest available checkpoint.

This is why I asked about setting the context to `u64::max` -- this should result in an error on transactions that attempt to make use of the context, because it's outside the available range.

Reply:
Resorting to u64::max for now, will follow up